src/osemgrep/cli_ci/Ci_subcommand.ml

```diff
@@ -894,27 +856,18 @@ let run_conf (caps : caps) (ci_conf : Ci_CLI.conf) : Exit_code.t =
         *)
         report_scan_completed ~blocking_findings ~blocking_rules
           ~non_blocking_findings ~non_blocking_rules;
-        let scan_id_opt = Option.map fst scan_config_opt in
         let app_block_override =
-          upload_findings ~dry_run caps depl_opt scan_id_opt prj_meta
-            blocking_findings filtered_rules cli_output
+          upload_findings ~dry_run caps' depl scan_id prj_meta blocking_findings
+            filtered_rules cli_output
         in
         let audit_mode = false in
         (* TODO: audit_mode = metadata.event_name in audit_on *)
         exit_code_of_blocking_findings ~audit_mode ~on:prj_meta.on
           ~app_block_override blocking_findings
   with
   | Error.Semgrep_error (_, ex) as e ->
-      (match (depl_opt, scan_config_opt) with
-      | Some (token, _), Some (scan_id, _scan_config) ->
-          let r =
-            match ex with
-            | None -> Exit_code.fatal ~__LOC__
-            | Some exit_code -> exit_code
-          in
-          let caps = Auth.cap_token_and_network token caps in
-          Semgrep_App.report_failure ~dry_run caps ~scan_id r
-      | _else -> ());
+      let r = ex ||| Exit_code.fatal ~__LOC__ in
```

Comment:
what does `|||` mean in OCaml?

Reply:
It's an operator I defined in Common.ml
```
let ( ||| ) a b =
  match a with
  | Some x -> x
  | None -> b
```

Reply:
ack – looks quite [similar](https://github.com/semgrep/semgrep/blob/9f07bade004c96b2f3fce583974552f608806570/src/osemgrep/configuring/Semgrep_envvars.ml#L44C5-L44C11) in flavor to our `env_or` function in `Semgrep_envvars.ml`

```
let env_or conv var default =
  match Sys.getenv_opt var with
  | None -> default
  | Some x -> conv x
```

Or the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) (`??`) in JavaScript